apluslms_roman/builder.py

```diff
@@ -54,26 +54,50 @@ def build(self, step_refs: list = None, clean_build=False):
         return result
 
 
+class BackendError(Exception):
+
+    def __init__(self, backend):
+        super().__init__()
+        self.backend = backend
+
+
 class Engine:
     def __init__(self, backend_class=None, settings=None):
+        backend_name = None
         if backend_class is None:
             if settings and 'backend' in settings:
                 backend_class = settings['backend']
+                if 'backends' in settings and backend_class in settings['backends']:
+                    backend_name = backend_class
+                    if 'type' in settings['backends'][backend_name]:
+                        backend_class = settings['backends'][backend_name]['type']
             else:
                 from .backends.docker import DockerBackend as backend_class
         if isinstance(backend_class, str):
             if '.' not in backend_class:
                 backend_class = BACKENDS.get(backend_class, backend_class)
-            backend_class = import_string(backend_class)
+            try:
+                backend_class = import_string(backend_class)
+            except ImportError:
+                raise BackendError(backend_class)
         self._backend_class = backend_class
 
-        name = getattr(backend_class, 'name', None) or backend_class.__name__.lower()
+        name = getattr(backend_class, 'name', None) or backend_class.__name__
         env_prefix = name.upper() + '_'
-        env = {k: v for k, v in environ.items() if k.startswith(env_prefix)}
-        if settings:
-            for k, v in settings.get(name, {}).items():
-                if v is not None and v != '':
-                    env[env_prefix + k.replace('-', '_').upper()] = v
+        env = {}
```

Comment:
While rewriting this part, we should rename this to `context` or `options` (backend options). PR #8 includes rewrite of how docker backend takes this env/context and that should be taken in consideration here. The idea is to replace some environment variables with just dictionary of options. The data is used here https://github.com/apluslms/roman/blob/7ae46077a67b94d5e80b2cd8b73d8c6932ffa285/apluslms_roman/backends/docker.py#L27. 

This would mean that when we are reading values from the environment, we would strip the `env_prefix`. PR #8 includes also some environment parsing related to this.

Reply:
Renamed `env` to `options` in some places, at least.